src/main/kotlin/app/configuration/JobCompletionNotificationListener.kt

```diff
@@ -13,6 +14,8 @@ class JobCompletionNotificationListener(private val writer: ItemWriter<String>):
     override fun afterJob(jobExecution: JobExecution) {
         if (writer is DirectoryWriter) {
             writer.writeOutput()
+        } else if (writer is S3DirectoryWriter) {
```

Comment:
if (validWriter(writer)) {}

Then a method than checks what a valid writer is? Or even just one if statement with an 'or'?

Reply:
As they do the same thing anyway.

Reply:
We can now, yes